currency_rate_update/models/res_currency_rate_provider.py

```diff
@@ -156,8 +157,13 @@ def _update(self, date_from, date_to, newest_only=False):
                 continue
 
             if not data:
-                # Try again if there is no data yet
+                # Try again if there is no data yet up to set number of times.
+                provider.no_data_occurrence += 1
+                if provider.no_data_occurrence > provider.allow_no_data_occurrence:
+                    provider.next_run = fields.Date.context_today(self)
+                    provider.no_data_occurrence = 0
```

Comment:
```suggestion
                if is_sheduled and provider.retry > provider.max_retry:
                    provider.next_run = fields.Date.context_today(self)
                    provider.retry = 0
```

Add the following at the beginning of the loop:

```
if is_scheduled:
    provider.retry += 1
```